widget/list.go

```diff
@@ -176,6 +177,15 @@ func (l *listRenderer) getItem() fyne.CanvasObject {
 }
 
 func (l *listRenderer) offsetChanged() {
+	if l.list.Length() == 0 {
```

Comment:
If the offset of the scroller is not 0 and then the list is cleared and refreshed, the offset of the scroller changes from X to 0.  This is not due to a scroll but an adjustment due to the scroller's child's min size changing.  Makes sense to me.

Reply:
If it is triggered by other code then why not put it there? 
Perhaps this deserves to be a different method like "resetScroll" and it's called when needed rather than in this scroll callback?

Reply:
I'm not sure why you think this is a scroll callback?  The callback is called offsetChanged - which happens when the scroller alerts us that the offset has changed - which is reasonably the case when the min size of the scroller content changes?  It makes sense for me to reset our offset tracking code to 0 in the offset changing callback when the offset changing callback is called with 0.  Would it make it more understandable if the code was if l.scroller.Offset.Y == 0 instead of l.list.Length() == 0?

Reply:
Moved variable resetting to the Length == 0 check inside of renderer's Layout called by Refresh.